src/services/openai.ts

```diff
@@ -84,67 +89,100 @@ export class OpenAIService {
 ${Object.entries(deckConfig.responseFields)
   .map(
     ([field, config]) =>
-      `- ${field}: ${config.description}${config.required ? " (required)" : " (optional)"}`,
+      `- ${field}: ${config.description}${config.required ? " (required)" : " (optional)"} ${config.locale ? `(locale: ${config.locale})` : ""}`,
   )
-  .join("\n")}`;
+  .join("\n")}
+  
+  Required fields must be present in the response.`;
   }
 
   async generateContent(
     card: Card,
     deckConfig: DeckConfig,
   ): Promise<DynamicContent> {
-    try {
-      // Validate fields before processing
-      this.validateFields(card, deckConfig);
-
-      // Replace field references in prompt
-      const processedPrompt = this.replaceFieldReferences(
-        deckConfig.prompt,
-        card,
-        deckConfig,
-      );
-
-      const fullPrompt = `${processedPrompt}\n\n${this.generateResponseFormat(deckConfig)}`;
-      this.debugLog(fullPrompt);
-
-      const completion = await this.client.chat.completions.create({
-        model: this.chatModel,
-        messages: [{ role: "user", content: fullPrompt }],
-        response_format: { type: "json_object" },
-      });
+    while (true) {
```

Comment:
if some field is missing, regenerate the content.

Reply:
Let's give it a max retry attempt limit

Reply:
sure thing 
https://github.com/Yukaii/gakuon/pull/107/commits/6df39fb8ab145854f64b5dfb3577a59a2ffcf1f2